skaha/src/main/java/org/opencadc/skaha/session/GetAction.java

```diff
@@ -176,86 +179,125 @@ private ResourceStats getResourceStats() throws Exception {
-                    totalNodeCPUCores = totalNodeCPUCores + cpuCore;
+                String nodeName = "";
+                int nodeCPUCores = 0;
+                for (String line : lines) {
+                    String[] parts = line.split("\\s+");
+                    if (nodeName.equals(parts[0])) {
+                        nodeCPUCores = nodeCPUCores + Integer.parseInt(parts[2]);
+                    } else {
+                        if (nodeName.length() > 0) {
+                            // processing first line of a subsequent nodeName
+                            nodeToCoresMap.put(nodeName, nodeCPUCores);
+                            nodeName = parts[0];
+                            nodeCPUCores = Integer.parseInt(parts[2]);
+                        } else {
+                            // processing first line of first nodeName
+                            nodeName = parts[0];
+                            nodeCPUCores = Integer.parseInt(parts[2]);
+                        }
+                    }
                 }
                 
-                return totalNodeCPUCores;
+                // processing last line of the last nodeName
+                nodeToCoresMap.put(nodeName, nodeCPUCores);
             }
         }
         
-        return 0;
+        return nodeToCoresMap;
     }
 
-    private int[] getAvailableResources(String nodeName, String k8sNamespace) throws Exception {
-        int resources[] = new int[2];
-        String getCPUCoresCmd = "kubectl -n " + k8sNamespace + " describe node " + nodeName;
-        String nodeCPUCores = execute(getCPUCoresCmd.split(" "));
-        if (nodeCPUCores != null) {
-            String[] lines = nodeCPUCores.split("\n");
-            boolean hasCores = false;
-            boolean hasRAM = false;
-            for (String line : lines) {
-                if (!hasCores && line.indexOf("cpu:") >= 0) {
-                    String[] parts = line.split(":");
-                    // number of cores from "Capabity.cpu"
-                    int cores = Integer.parseInt(parts[1].trim());
-                    log.debug("Available CPU cores in node " + nodeName + ": " + cores);
-                    resources[0] = cores;
-                    hasCores = true;
+    private Map<String, int[]> getAvailableResources(String k8sNamespace) throws Exception {
+        String getAvailableResourcesCmd = "kubectl -n " + k8sNamespace + " describe nodes ";
+        String rawResources = execute(getAvailableResourcesCmd.split(" "));
+        Map<String, int[]> nodeToResourcesMap = new HashMap<String, int[]>();
+        if (StringUtil.hasLength(rawResources)) {
+            String[] lines = rawResources.split("\n");
+            if (lines.length > 0) {
+                List<String> keywords = Arrays.asList("Name:", "Capacity:", "cpu:", "memory:");
+                String nodeName = "";
+                boolean hasName = false;
+                boolean hasCapacity = false;
```

Comment:
What is 'capacity'?  memory?

Reply:
They are in the output of "kw describe nodes". As we process each line in the output, we want to know where are in the output. We want 'Capabity.cpu' and 'Capabity.memory'; however we do not want 'Allocatable.cpu' and 'Allocatable.memory'.